File: backend/Entities/Match.cs
using backend.Enums;

namespace backend.Entities;

public class Match
{
    public int Id { get; set; }
    public int TeamAId { get; set; }
    public Team TeamA { get; set; } = null!;
    public int TeamBId { get; set; }
    public Team TeamB { get; set; } = null!;
    public int TournamentId { get; set; }
    public Tournament Tournament { get; set; } = null!;
    public MatchStatus Status { get; set; }
    public MatchFormat Format { get; set; }
    public DateTime StartTime { get; set; }
    public DateTime? EndTime { get; set; }
    public int ScoreA { get; set; }
    public int ScoreB { get; set; }
    public DateTime CreatedAt { get; set; }

    public ICollection<MatchMap> Maps { get; set; } = [];
}

File: frontend/src/types/Match.ts
import { Team } from '@/types/Team';
import { Map } from '@/types/Map';

export type MatchStatus = 'Scheduled' | 'Live' | 'Finished';

export type MatchFormat = 'Bo1' | 'Bo3' | 'Bo5';

export interface Match {
  id: number;
  teamA: Team;
  teamB: Team;
  tournamentId: number;
  tournamentName: string;
  region: string;
  status: MatchStatus;
  format: MatchFormat;
  startTime: string;
  endTime: string | null;
  scoreA: number;
  scoreB: number;
  maps: Map[];
}
